pkg/git/git.go

```diff
@@ -0,0 +1,57 @@
+// Copyright The Shipwright Contributors
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package git
+
+import (
+	"fmt"
+
+	"github.com/go-git/go-git/v5/config"
+	"github.com/go-git/go-git/v5/plumbing/transport"
+	"github.com/go-git/go-git/v5/storage/memory"
+
+	gogitv5 "github.com/go-git/go-git/v5"
+)
+
+const (
+	defaultRemote = "origin"
+	httpsProtocol = "https"
+	fileProtocol  = "file"
+	gitProtocol   = "ssh"
+)
+
+// ValidateGitURLExists validate if a source URL exists or not
+// Note: We have an upcoming PR for the Build Status, where we
+// intend to define a single Status.Reason in the form of 'remoteRepositoryUnreachable',
+// where the Status.Message will contain the longer text, like 'invalid source url
+func ValidateGitURLExists(urlPath string) error {
+
+	endpoint, err := transport.NewEndpoint(urlPath)
+	if err != nil {
+		return err
+	}
+
+	switch endpoint.Protocol {
+	case httpsProtocol:
```

Comment:
yeah correlating with my comment in the doc section, `http` is a valid, if not primary, use case, that should be easy to add

and this would be the spot, correct?

Reply:
I'd suggest that it should log out that it's insecurely accessing the repository when using plain HTTP.

It _could_ be a typo.

Reply:
I added it. Thanks for the feedback. @bigkevmcd somehow I cannot re-request a review to you, not sure why.